src/System.Reflection.Metadata/src/System/Reflection/Throw.cs

```diff
@@ -2,118 +2,192 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
 using System.Reflection.Metadata.Ecma335;
 using System.Runtime.CompilerServices;
 
 namespace System.Reflection
 {
     // This file defines an internal class used to throw exceptions. The main purpose is to reduce code size.
     // Also it improves the likelihood that callers will be inlined.
+    // Note however that Watson bucketing will merge all crashe reports with the same stack trace.
```

Comment:
1. nit: typo s/crashe/crash

2.  don't understand the comment: is it that we get worse buckets due to the same top frame? Same "stack trace" doesn't imply any downside to this approach.